### PR TITLE
[Woo POS] Ensure Product Names are Displayed Correctly

### LINKED ISSUE #12550: [Woo POS] Orders created in POS don’t show product names when reviewed in IPP later on
* Complete a transaction in POS
* Go to orders
* Review the details of the order
* Result: The right number of products and their respective quantities/prices are correct, but the names of the products aren’t passed through

p91TBi-c2x-p2#comment-13033

### PR DESCRIPTION
**Pull request by backwardstruck**
Closes: #12550  

### Description  

Addresses an issue where orders created in POS do not display product names when reviewed in the regular IPP flow. The problem was due to the `Order.Item` instances not being fully populated with product details. 

The fix involves updating the `WooPosTotalsRepository#createOrderWithProducts` method to fetch complete product details before creating order items, ensuring all necessary fields are filled.

### Steps to reproduce  

1. Complete a transaction in Woo POS with a few simple products.
2. Navigate to the orders section of the main app.
3. Review the details of the order.
4. We should have the same fields filled (including the item name) in the order as we do when creating the order through IPP:
```
Item(itemId=1332, productId=15, name=Garment-dyed Super Soft Slub Cotton Hoodie in Ash, price=70, sku=, quantity=1.0, subtotal=70.00, totalTax=1.05, total=70.00, variationId=0, attributesList=[], parent=null, configuration=null, configurationKey=null)  
```

### Testing information  

Devices Used:
- tablet emulator

### The tests that have been performed  

1. **Basic Transaction Test**:
   - Completed a transaction in Woo POS and verified that the product names appear correctly in the order review in IPP.
   
2. **Edge Case Test – Multiple Products**:
   - Added multiple products to the cart and completed the transaction.
   - Confirmed that all product names, prices, and other details are visible in the IPP.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsRepositoryTest.kt [hunk 1]
```diff
@@ -57,17 +95,49 @@ class WooPosTotalsRepositoryTest {
...
+
+        // THEN
+        val orderCapture = argumentCaptor<Order>()
+        verify(orderCreateEditRepository).createOrUpdateOrder(
+            orderCapture.capture(),
+            eq("")
+        )
+
+        assertThat(orderCapture.lastValue.items.size).isEqualTo(1)
+        assertThat(orderCapture.lastValue.items[0].name).isEqualTo(product1.name)
```
Comment (backwardstruck) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsRepositoryTest.kt:124]: Test this specific scenario we fixed.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/237">20.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by samiuelson — APPROVED** [review]
LGTM :shipit:
